diff --git a/cmd/server-main.go b/cmd/server-main.go
index ade86cc..d34e397 100644
--- a/cmd/server-main.go
+++ b/cmd/server-main.go
@@ -805,8 +805,7 @@ func serverMain(ctx *cli.Context) {
 	// Initialize KMS configuration
 	bootstrapTrace("handleKMSConfig", handleKMSConfig)
 
-	// Load the root credentials from the shell environment or from
-	// the config file if not defined, set the default one.
+	// Load explicit root credentials or generate them using the configured KMS.
 	bootstrapTrace("rootCredentials", func() {
 		cred := loadRootCredentials()
 		if !cred.IsValid() && (env.Get(api.EnvAPIRootAccess, config.EnableOn) == config.EnableOff) {
@@ -815,8 +814,8 @@ func serverMain(ctx *cli.Context) {
 			cred = autoGenerateRootCredentials()
 		}
 
-		if !cred.IsValid() {
-			cred = auth.DefaultCredentials
+		if !cred.IsValid() || cred.Equal(auth.DefaultCredentials) {
+			logger.Fatal(errors.New("explicit, non-default root credentials are required"), "Unable to start MinIO")
 		}
 
 		var err error
